pkg/pkgrepository/package_repo_app.go

```diff
@@ -108,7 +115,132 @@ metadata:
+    kapp.k14s.io/create-strategy: "fallback-on-update-or-noop"`, pkgRepository.Namespace, pkgRepository.Name),
+					},
+				},
+			},
+		}, {
+			Ytt: &kcv1alpha1.AppTemplateYtt{
+				Paths: []string{"-"},
+				Inline: &kcv1alpha1.AppFetchInline{
+					Paths: map[string]string{
+						//   rebase rule to allow multiple repositories to expose identical packages.
+						"noop-on-identical-packages.yml": `
+---
+apiVersion: kapp.k14s.io/v1alpha1
+kind: Config
+rebaseRules:
+- ytt:
+    overlayContractV1:
+      overlay.yml: |
+        #@ load("@ytt:data", "data")
+        #@ load("@ytt:yaml", "yaml")
+        #@ load("@ytt:json", "json")
+        #@ load("@ytt:overlay", "overlay")
+        #@ load("@ytt:struct", "struct")
+
+        #@ def get_rev(annotations):
+        #@   if hasattr(annotations, "packaging.carvel.dev/revision"):
+        #@     return int(annotations["packaging.carvel.dev/revision"])
+        #@   else:
+        #@     return -1
+        #@   end
+        #@ end
+        #@ if not hasattr(data.values.existing.metadata.annotations, "packaging.carvel.dev/package-repository-ref"):
+        #@   msg = "Error: cannot overwrite package " + data.values.existing.metadata.name + " because it was not created by a package repository."
+        #@   print(msg)
+        #@   fail(msg)
+        #@ end
+
+        #@ def filter(d, s):
+        #@   return {x: v for x, v in d.items() if not x in s}
+        #@ end
+
+        #! TODO: next person who adds an annotation to the set of all kapp-controller annotations is gonna have a bad time.
+        #@ def annotations_are_identical(existing_anns, new_anns):
+        #@   carvel_anns = set(["kapp.k14s.io/disable-original",
+        #@                      "kapp.k14s.io/disable-wait",
+        #@                      "packaging.carvel.dev/package-repository-ref",
+        #@                      "kapp.k14s.io/identity",
+        #@                      "kapp.k14s.io/create-strategy"])
+        #@   return filter(existing_anns, carvel_anns) == filter(new_anns, carvel_anns)
+        #@ end
+
+        #@ def labels_are_identical(existing_labels, new_labels):
+        #@   carvel_labels = set(["kapp.k14s.io/app", "kapp.k14s.io/association"])
+        #@   return filter(existing_labels, carvel_labels) == filter(new_labels, carvel_labels)
+        #@ end
+        #@
+        #@ def specs_are_identical(existing_spec, new_spec, kind):
+        #@   ex = existing_spec
+        #@   nw = new_spec
+        #@
+        #@   for attr in nw.keys():
+        #@     if not attr in ex:
+        #@       return False, "adding spec."+attr
+        #@     end
+        #@   end
+        #@
+        #@   for attr in ex.keys():
+        #@     if not attr in nw:
+        #@       return False, "missing spec." + attr
+        #@     end
+        #@     if nw[attr] != ex[attr]:
+        #@       return False, "mismatch in spec." + attr
+        #@     end
+        #@   end
+        #@   return True, ""
+        #@ end
```

Comment:
note from earlier convo: can we switch to using ==, and only do special casing when hinting at differences?